Sources/ArcGISToolkit/FloorFilter/FloorFilterViewController.swift

```diff
@@ -0,0 +1,534 @@
+        state = FloorFilterState.initiallyCollapsed
+        updateViewsVisibilityForState(state: state)
+    }
+    
+    /// Variables for styling the Floor Filter View
+    public var fontSize: CGFloat = 14.0
+    public var fontName: String = "Avenir"
+    public var selectionColor: UIColor = UIColor(hexString: "#C7EAFF")
+    public var backgroundColor: UIColor = UIColor.systemGray6
+    public var selectedTextColor: UIColor = UIColor(hexString: "#004874")
+    public var unselectedTextColor: UIColor = UIColor.label
+    
+    /// Floor Filter UI Elements and Constraints
+    @IBOutlet var floorFilterView: UIView!
+    @IBOutlet weak var siteBtn: UIButton!
+    @IBOutlet weak var levelsTableView: UITableView!
+    @IBOutlet weak var closeBtn: UIButton!
+    @IBOutlet weak var floorFilterStackView: UIStackView!
+    @IBOutlet weak var closeBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnHeight: NSLayoutConstraint!
+    @IBOutlet weak var tableViewHeight: NSLayoutConstraint!
+    @IBOutlet weak var closeBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var siteBtnWidth: NSLayoutConstraint!
+    @IBOutlet weak var levelCellWidth: NSLayoutConstraint!
+    
+    /// Floor Filter view styles variables
+    /// These are the default, but will get updated with values that are provided during initialization
+    private var buttonHeight: CGFloat = 35
+    private var buttonWidth: CGFloat = 40
+    private var maxDisplayLevels: Int = 3
+  
+    private var delegate: FloorFilterViewControllerDelegate?
+    private var viewModel = FloorFilterViewModel()
+    
+    /// State of the visibility of the Floor Filter
+    private enum FloorFilterState {
+        case initiallyCollapsed
+        case partiallyExpanded
+        case fullyExpanded
+    }
+    private var state = FloorFilterState.initiallyCollapsed
+    
+    private var floorManager: AGSFloorManager?
+    
+    /// GeoView that the Floor Filter is rendered on
+    /// For Version 1 only MapView (2D) is supported to render the Floor Filter
+    private var geoView: AGSGeoView? {
+        didSet {
+            if geoView != nil {
+                if let mapView = geoView as? AGSMapView {
+                    mapView.map?.load { [weak self] (_) in
+                        guard let self = self else { return }
+                        self.viewModel.mapView = mapView
+                        self.viewModel.map = mapView.map
+                        self.floorManager = mapView.map?.floorManager
+                        self.initializeFloorManager()
+                    }
+                }
+            }
+        }
+    }
+    
+    /// Static method that will be used to initialize the Floor Filter View and attach it as a SubView
+    public static func makeFloorFilterView(
+        geoView: AGSGeoView?,
+        buttonWidth: CGFloat = 50,
+        buttonHeight: CGFloat = 50,
+        maxDisplayLevels: Int = 3,
+        style: Style = .expandUp
+    ) -> FloorFilterViewController? {
```

Comment:
@philium thanks for the observations. Agree that it would be better with its own .xib file. However, I was following the design for the `LegendViewController` component for my initial implementation with the static method to create the ViewController. I'd like to refactor the FloorFilterViewController into its own .xib file as an enhancement once this initial implementation gets merged.